### DISCUSSION TITLE
When use lgf.compute_structure_environments module. And error " 'MaterialsDoc' object is not subscriptable

### DISCUSSION THREAD

**Discussion by Aiartanis** [materialsproject/pymatgen#2918] [Q&A] [answered]
# Get the StructureEnvironments
When I follow the Supporting infomation.   i meet the problem. Please help==。

se = lgf.compute_structure_environments(maximum_distance_factor=1.41, only_cations=False)

![image](https://user-images.githubusercontent.com/95071614/228899005-c286b687-2e85-4f6d-825a-cc3e09389fff.png)

**Comment by janosh** [comment, marked as answer]
Please provide a runnable code snippet that triggers the error. From your screenshot, it looks like you're passing in a `MPRester` `MaterialsDoc` rather than a `Structure` instance to `LocalGeometryFinder`. If your `MaterialsDoc` contains a structure, try passing in `doc['structure']` instead.

**Reply by Aiartanis** [reply]
i am very sorry, i tried the cif and it works。
maybe the link from Materials Project is failed
thank you Very much

**Reply by janosh** [reply]
The object returned by `mpr.materials.get_data_by_id()` is a `MaterialsDoc`, not a `Structure`. If you access it's `structure` key rather than the passing the doc itself into `lgf.setup_structure()`, I think it should work.

**Reply by Aiartanis** [reply]
oh……maybe i need try again ☺☺
Thank you !!
